reference-adapter/ifix-department-entity-service/src/main/java/org/egov/repository/criteriaBuilder/DepartmentQueryCriteria.java

```diff
@@ -0,0 +1,96 @@
+package org.egov.repository.criteriaBuilder;
+
+import org.egov.web.models.persist.DepartmentEntity;
+import org.egov.web.models.persist.DepartmentHierarchyLevel;
+import org.springframework.util.StringUtils;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+
+public class DepartmentQueryCriteria {
+
+    private StringBuilder criteriaQuery = new StringBuilder();
+
+    private DepartmentQueryCriteria() {}
+
+    public DepartmentQueryCriteria(StringBuilder criteriaQuery) {
+        this.criteriaQuery = criteriaQuery;
+    }
+
+    public static DepartmentQueryCriteria builder(Class<? extends Object> entityClass) {
+
+        if (DepartmentEntity.class.getTypeName().equals(entityClass.getTypeName())) {
+            return new DepartmentQueryCriteria(new StringBuilder("SELECT * FROM department_entity"));
```

Comment:
keep the base query saved in static const string field

Reply:
Done